robottelo/vm.py

```diff
@@ -114,9 +115,12 @@ def create(self):
         if self.image_dir is not None:
             command_args.append('-p {image_dir}')
 
+        if self._domain is None:
+            self._domain = self.fetch_domain()
```

Comment:
fetch_domain could return an empty string if the provisioning server does not have a FQDN. We should raise `VirtualMachineError` if it happens.